hnn_core/viz.py

```diff
@@ -242,29 +244,45 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from .dipole import Dipole
+    from .dipole import Dipole, average_dipoles
 
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
     if isinstance(dpl, Dipole):
         dpl = [dpl]
+        average = False
+    else:
+        # add average plot when we have dpl>1 and `average` is specified.
+        if average:
+            dpl = dpl + [average_dipoles(dpl)]
 
+    linewidth = 1.5
     scale_applied = dpl[0].scale_applied
-    for dpl_trial in dpl:
+    for idx, dpl_trial in enumerate(dpl):
```

Comment:
I'm trying to resist the temptation to avoid micro-optimizing your code but it's hard for me ;-)
```suggestion
    for dpl_trial in dpl[:-1]:
```
now if the last element of the `dpl` list is always the average, then you can move the:

```py
if idx == len(dpl) - 1 and average:
```

outside of the loop and into the:

```py
  if average:
        ax.legend()
```

Reply:
You mean also to move

```Python
            # extract scaled data and times
            data, times = _get_plot_data_trange(dpl_trial.times,
                                                dpl_trial.data[layer], tmin,
                                                tmax)
            if decim is not None:
                data, times = _decimate_plot_data(decim, data, times)
```

out of the loop and put into the if average?

Reply:
indeed, sorry I didn't realize that would also need to change. I think it's fine as it is. I pushed some cosmetic changes. Made the default color to be "black" rather than do a color cycle and tweaked the linewidth a bit.